src/main/scala/com/fulcrumgenomics/umi/CallMolecularConsensusReads.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) $year Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.umi.ConsensusCallerOptions._
+import com.fulcrumgenomics.util.LogDouble._
+import com.fulcrumgenomics.util.{LogDouble, ProgressLogger}
+import dagr.commons.CommonsDef.PathToBam
+import dagr.commons.io.Io
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import dagr.sopt.cmdline.ValidationException
+import htsjdk.samtools._
+import htsjdk.samtools.util.CloserUtil
+
+import scala.collection.JavaConverters._
+
+
+@clp(description =
+  """
+    |Calls consensus sequences from reads with the same UMI identifier.
+    |
+    |This tool assumes that reads with the same identifier are grouped together (consecutive in the file).
+  """,
+  group = ClpGroups.Umi)
+class CallMolecularConsensusReads
+( @arg(flag="i", doc="The input SAM or BAM file.") val input: PathToBam,
+  @arg(flag="o", doc="Output SAM or BAM file to write consensus reads.") val output: PathToBam,
+  @arg(flag="r", doc="Output SAM or BAM file to write reads not used.") val rejects: PathToBam,
+  @arg(flag="t", doc="The SAM attribute with the unique UMI identifier.") val tag: String = DefaultTag,
```

Comment:
I don't think this is strictly a UMI tag - it's a `UM` tag ;)  But seriously, in `GroupReadsByUmi` I'm emitting a tag that is a unique identifier for an original molecule, file-wide, i.e. it's broader scoped than a corrected UMI.